install/kubernetes/helm/istio/templates/configmap.yaml

```diff
@@ -88,12 +88,12 @@ data:
       controlPlaneAuthPolicy: MUTUAL_TLS
       #
       # Address where istio Pilot service is running
-      discoveryAddress: istio-pilot.{{ .Release.Namespace }}:15005
+      discoveryAddress: istio-pilot.{{ .Release.Namespace }}:15011
```

Comment:
@sdake , how did you test this? the proxyv2 will not work with port 15005 and 15007?

Reply:
was tested with the e2e tests.  I do not know if proxyv2 will not work with port 15005 or 15007.  This problem was found in a different review.

Reply:
@sdake can you please show me the review? I want to do some test, but stuck at https://github.com/istio/istio/issues/6460

Reply:
@sdake it works with both port 15007 and 15010 without auth.

@rshriram any recommendation for this?

Reply:
Obtain the values from the Envoy v2 transition.yaml. Also I think Costin has some ports fixed in the bootstrap/server.go

Reply:
@rshriram as we need to configure this in configmap, seems there is no good way to get it from transition.yaml and put it to configmap?

@costinm what kind of ports did you update for bootstrap/server.go?

Reply:
It looks like `pkg/bootstrap/bootstrap_config.go :: WriteBootstrap()` is always rendering a grpc config and overrides the port in discoveryAddress:

```
	h, p, err := GetHostPort("Discovery", config.DiscoveryAddress)
	if err != nil {
		return "", err
	}
	StoreHostPort(h, p, "pilot_address", opts)

	// Default values for the grpc address.
	// TODO: take over the DiscoveryAddress or add a separate mesh config option
	// Default value
	grpcPort := "15010"
	if config.ControlPlaneAuthPolicy == meshconfig.AuthenticationPolicy_MUTUAL_TLS {
		grpcPort = "15011"
	}
	grpcHost := h // Use pilot host
```

so I think our helm chart change doesn't matter :-)

Reply:
Please fix this TODO. We had to find the hard way, too, since we could not understand why telemetry for pilot was missing, and this is the culprit.

Reply:
As this will be used to inject sidecar containers, so I think we may still need to update it.

I found that the porxyv1 is using `pilot_grpc_address` and `pilot_address`, and proxyv2 is only using `pilot_grpc_address` in envoy, then my question is why proxyv2 also works with 15005 and 15007 which is the `pilot_address`?

@kyessenov @costinm @rshriram comments for this?

Reply:
@gyliu513 -- it looks like that code snippet I pasted above is ignoring the port setting for pilot_address (discoveryAddress).  So, 15005 and 15007 aren't actually used by envoy sidecars anymore no matter what you set in the mesh configmap.

You can see confirmation in your setup via checking out the envoy bootstrap config in one of your app's sidecars (`istio-proxy` container).  The following is from a setup with the default helm chart rendered configmap with the 15005 port setting in discoveryAddress:

`cat /etc/istio/proxy/envoy-rev0.json`
```
  "dynamic_resources": {
    "lds_config": {
        "ads": {}
    },
    "cds_config": {
        "ads": {}
    },
    "ads_config": {
      "api_type": "GRPC",
      "refresh_delay": {"seconds": 10, "nanos": 0},
      "grpc_services": [
        {
          "envoy_grpc": {
            "cluster_name": "xds-grpc"
          }
        }
      ]
    }
   ...
  "static_resources": {
    "clusters": [
    {
    "name": "xds-grpc",
    "type": "STRICT_DNS",
    "connect_timeout": {"seconds": 1, "nanos": 0},
    "lb_policy": "ROUND_ROBIN",

    "hosts": [
    {
    "socket_address": {"address": "istio-pilot.istio-system", "port_value": 15010}
    }
   ...
```

Reply:
@gyliu513 - this review exposed this problem: https://github.com/istio/istio/pull/6366

Reply:
@tiswanso Yes, this is also my checking result. But my question is, during my test, if the injected container using 15005 or 15007 with proxyv2, it also works fine, my thinking is that those two ports should be failed?

Reply:
yes - the reason it works when you inject with 15005 or 15007 is that bootstrap.go always hardcodes to specific ports for proxyv2.  The part I don't understand is why proxyv1 works at all, however it doesn't matter as proxyv1 is going away in 1.0.

Reply:
@rshriram after un-hardcoding the port, the gate implodes.  Any insight?

Reply:
Gateways don’t read the global sidecar injector config. So if you want to make pilot port parameterizable, add option to global config and make pilot and pilot agent read it during startup

Reply:
@gyliu513 -- that's the behavior of the code in [1] & the thing @sdake is trying to change with these latest versions of the review.  At the moment, no matter what port is passed in `discoveryAddress` to the istio-proxy (sidecar) it will configure envoy ADS over port 15010 or 15011 (mTLS).  

[1] https://github.com/istio/istio/pull/6452/commits/6fc1a85ef5112678797e7b358efa9def3bd907a6#r197216278

Reply:
@tiswanso so seems the parameter `discoveryAddress` in injected in sidecar container has no use? Then why do we need to set this parameter?